llama_index/callbacks/base.py

```diff
@@ -126,6 +144,21 @@ def set_handlers(self, handlers: List[BaseCallbackHandler]) -> None:
         """Set handlers as the only handlers on the callback manager."""
         self.handlers = handlers
 
+    @contextmanager

```

Comment:
very nit: I find it slightly surprising that the caller generally have to explicitly call `on_end` to log the end payload, whereas the `on_end` is implicit.

Could be helpful to write this down in the docstring.

Reply:
I can add it to the docstring. The only reason they have to (and it's optional) is because the payload might not be ready until the end of the event, and we need to pick it up somehow